app/controllers/storytime/posts_controller.rb

```diff
@@ -48,5 +50,14 @@ def show
       end
     end
 
+    private
+
+      def get_search_type
+        if params[:type]
+          "#{params[:type].camelize}".constantize
```

Comment:
This needs to be protected in some way, so we aren't calling constantize on any & all input a user might give. Something like:
```
"#{params[:type].camelize}".constantize if legal_search_types.include?(params[:type])
```